packages/nuxt/src/pages/utils.ts

```diff
@@ -94,6 +101,12 @@ export function generateRoutesFromFiles (files: string[], pagesDir: string): Nux
   return prepareRoutes(routes)
 }
 
+function getRouteName (file: string) {
+  const regExp = /definePageMeta\(\s*{[^}]*name:\s*(["'`])((?:[^"'`\\]|\\.)+?)\1/g
+  const match = regExp.exec(file)
```

Comment:
slightly better perf to hoist regexp out of function, and I think also we don't want it to be global.

```suggestion
const PAGE_META_NAME_RE = /definePageMeta\(\s*{[^}]*name:\s*(["'`])((?:[^"'`\\]|\\.)+?)\1/
function getRouteName (file: string) {
  const match = PAGE_META_NAME_RE.exec(file)
```

Reply:
Having said that, I do have questions about whether this might yield false positives - I'll need to do some testing